### ISSUE TITLE
Including arbitrary data in schema json

### ISSUE THREAD

**Issue by catsandkites** [tiangolo/fastapi#1969]
### First check

* [X] I added a very descriptive title to this issue.
* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [X] I already read and followed all the tutorial in the docs and didn't find an answer.
* [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [X] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

```Python
from fastapi import FastAPI, Request
from pydantic import BaseModel, Field

class ExampleModel(BaseModel):
    title:       str
    overview:    str = Field("", extra_arg="testtest")

    class Config:
        schema_extra = {
            "example": {
                "test": "test",
            },
            "notexample": {
                "test": "test",
            }
        }

app = FastAPI()

@app.get("/testroute/", response_model=ExampleModel)
async def testroute():
    
    print(ExampleModel.schema())
    print(ExampleModel.schema_json())
    print(app.openapi())

    return {'title': 'test', 'overview': 'test'}
```

### Description

* Open the browser and call the endpoint `/testroute/`.
* Or visit `/openapi.json`
* The extra argument (test_arg) to the overview Field is returned by `schema()` and `schema_json()`.
* However the extra argument (test_arg) to the overview Field is not returned in the OpenAPI schema.
* The example key in schema_extra is returned in the OpenAPI schema.
* However the notexample key in schema_extra is not returned in the OpenAPI schema.

The docs at [field-additional-arguments](https://fastapi.tiangolo.com/tutorial/schema-extra-example/#field-additional-arguments) say: "you can also declare extra info for the JSON Schema by passing any other arbitrary arguments to the function" and "The same way, you could add your own custom extra info that would be added to the JSON Schema for each model, for example to customize a frontend user interface, etc." which is what I am attempting to do.

The extra information is present when getting the schema from the pydantic class itself, but once it's passed through fastapi the extra information has been removed. It seems like only keys that fastapi/openapi is expecting as exported, however doesn't that contrast with documentation statements above?

Thank you.

### Environment

* OS: Ubuntu: 18.04
* Python: 3.6.9
* FastAPI Version: 0.60.1

**Comment by thomasleveil** [comment]
I was also misleaded by the doc and hoped to have the following working : 

```python
@app.get("/something")
def something(
        cache_control: str = Header(
            default=None, 
            examples={
                "no-cache": {"value": "no-cache"},
                "": {value": ""},
                "no-store": {value": "no-store"}
            }
        )
    ):
    ...
              
``

**Comment by catsandkites** [comment]
A work around in my case was to implement the[ Extending OpenAPI](https://fastapi.tiangolo.com/advanced/extending-openapi/) example and brute force my values into the dictionary at that point:

`openapi_schema["components"]["schemas"]["blah"]["properties"]["field"]["custom"] = "data"`

This works, however feels messy. It would be nice to understand why exactly custom field information disappears as the docs do seem misleading about this point.

**Comment by mr-bjerre** [comment]
Shouldn't this be marked as a bug rather than a question? I have the same problem and according to the docs that you cited this should work.

**Comment by jakob-ed** [comment]
I agree that the docs are misleading. In case you're just looking to define additional fields in valid JSON Schema in the `schema_extra`, here's how it can be accomplished in your example:

```python
from typing import Dict, Any

from fastapi import FastAPI
from pydantic import BaseModel

class ExampleModel(BaseModel):
    title:       str
    overview:    str

    class Config:
        @staticmethod
        def schema_extra(schema: Dict[str, Any]) -> None:
            schema["example"] = {
                "test": "test",
            }
            schema["properties"]["notexample"] = {
                "title": "Not Example",
                "type": "integer"
            }

app = FastAPI()

@app.get("/testroute/", response_model=ExampleModel)
async def testroute():

    print(ExampleModel.schema())
    print(ExampleModel.schema_json())
    print(app.openapi())

    return {'title': 'test', 'overview': 'test'}
```

The new field will then show up in the docs and openapi.json. Using a callable for `schema_extra` so that an entry to `properties` can be added instead of overwriting existing ones is documented in the [Pydantic docs](https://pydantic-docs.helpmanual.io/usage/schema#__code_11).

**Comment by LukasMasuch** [comment]
Even so Pydantic allows custom metadata in the JSON Schema, FastAPI seems to only allow properties defined in the [SchemaBase Model](https://github.com/tiangolo/fastapi/blob/master/fastapi/openapi/models.py#L82). All extra arguments are ignored at the creation of the OpenAPI schema [here](https://github.com/tiangolo/fastapi/blob/master/fastapi/openapi/utils.py#L377) since they are not part of the SchemaBase Model. 

Adding extra arguments to `properties` is a possible workaround since `properties` is [part of the SchemaBase Model](https://github.com/tiangolo/fastapi/blob/master/fastapi/openapi/models.py#L106). Unfortunately, this is not really compatible with Pydantic's support of extra arguments. So, I think this part of the documentation - `you can also declare extra info for the JSON Schema by passing any other arbitrary arguments to the function` - is not possible with the current FastAPI version.
